### PR TITLE
Add Hypertrace tracestate header

### PR DESCRIPTION
**Pull request by pavolloffay**
Add doc to define format of tracestate header to encode HT specific information

### CODE REVIEW

File: agent/tracestate.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+#### Capture
+
+The capture instructs agent what data should be captured. The agent MUST set the capture value
+if the value was not previously present.
+
+Value key: `cap`
+
+Possible values:
+* `0` - capture all request/response headers and payloads
```
Comment (davexroth) [inline agent/tracestate.md]: should we change the values here?

`cap == 1` reads more naturally that data capture has been enabled.

File: agent/tracestate.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+* `1` - do not capture payloads and headers except whitelisted ones
+
+An example:
+* `tracestate: hypertrace=cap:0`.
+
+##### Whitelisted headers
+
+The whitelisted headers are 
+* `Content-Type`
```
Comment (davexroth) [inline agent/tracestate.md:40]: the whitelisted headers should be configurable in each of the agents, with these two being the default.

File: agent/tracestate.md (hunk 1 above)
Comment (pavolloffay) [reply agent/tracestate.md]: +1

Comment (pavolloffay) [reply agent/tracestate.md]: my original thinking was that the "advanced" mode is the default.

File: agent/tracestate.md (hunk 2 above)
Comment (pavolloffay) [reply agent/tracestate.md:40]: This will be documented in the agent-config.

**Review by davexroth — APPROVED** [review]

File: agent/tracestate.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+Value key: `cap`
+
+Possible values:
+* `0` - do not capture payloads and headers except whitelisted ones
+* `1` - capture all request/response headers and payloads
+
+An example:
+* `tracestate: hypertrace=cap:0`
+
+##### Whitelisted headers
```
Comment (jcchavezs) [inline agent/tracestate.md:37]: Use allowlist

**Comment by jcchavezs** [comment]
Hi @pavolloffay could you please describe the use cases this would be useful for. In the way I interpret this we want to make hops to capture certain data based on a propagated value, is that the case? Under which cases we want to tell the next hops to capture A or B, also, how do one hop knows what is the allow list of headers for next hops?

**Comment by pavolloffay** [comment]
I have added more text. Please read it if it makes it clearer.

File: agent/tracestate.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+The Hypertrace tracestate vendor value can encode multiple values. The values are separated by `|`. 
+
+An example:
+* `tracestate: somevendor=opaqueValue1,hypertrace=key1:val1|key2:val2`
+
+#### Capture
+
+The capture feature allows coherently capturing data within a trace. For instance a sampler
+can make use of this capability to capture only some percentage of request and response payloads.
```
Comment (jcchavezs) [inline agent/tracestate.md:23]: I see you added this example but how exactly can we achieve this with `cap`?

Comment (pavolloffay) [reply agent/tracestate.md:23]: 1. sampler makes the decision and sets the cap value
2. instrumentation uses tracestate to capture values OR the data is dropped before reporting  OR there is a filter for adding a data to span

2. depends on the instrumentation API. In OTEL it seems that the instrumentation points will have to look at the tracestate to determine whether data should be captured or not.

**Review by jcchavezs — APPROVED** [review]

**Pull request merged**